core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -63,6 +62,31 @@ sealed abstract class UTXOSpendingInfo {
   def conditionalPath: ConditionalPath
 }
 
+/**
+  * Contains the information required to spend an unspent transaction output (UTXO)
+  * on a blockchain.
+  */
+sealed abstract class UTXOSpendingInfo extends UTXOSpendingInfoSingle {
+
+  /** All signers needed to spend from the output above */
+  def signers: Seq[Sign]
+
+  /** If a UTXOSpendingInfo is called where a UTXOSpendingInfoSingle is used,
+    * the first key is used.
+    * This is nice because it allows us to use things like P2PKHSpendingInfo in both places
+    * where full signing and single key singing happen.
+    */
+  override def signer: Sign = signers.head
+
+  def requiredSigs: Int
+
+  def toSingle(signerIndex: Int): UTXOSpendingInfoSingle
+
+  def toSingles: Vector[UTXOSpendingInfoSingle] = {
+    signers.indices.take(requiredSigs).toVector.map(toSingle)
```

Comment:
If we had a 2-of-3 multisig, this would only allow us to sign with the first 2 keys

Reply:
So actually it will allow you to sign with any two keys but if you happen to have all three then it will only use the first two in the `Vector` (which could still be any two). Will add a doc tho